packaging/macosx/4_make_hb_darktable_dmg.sh

```diff
@@ -19,6 +19,8 @@ cd "$buildDir"/
 # Generate symlink to applications folder for easier drag & drop within dmg
 ln -s /Applications package/ || true
 
+echo "Killing XProtect..."; sudo pkill -9 XProtect >/dev/null || true; # see https://github.com/actions/runner-images/issues/7522
```

Comment:
```suggestion
if [ -n "${GITHUB_RUN_ID}" ]; then
    # see https://github.com/actions/runner-images/issues/7522
    echo "Killing XProtect..."; sudo pkill -9 XProtect >/dev/null || true;
fi
```

Reply:
Here's what I read in the GitHub documentation:
`Because default environment variables are set by GitHub and not defined in a workflow, they are not accessible through the env context. `
That is, we cannot check whether this variable is in the environment.

Reply:
Ok, then we need to set our own unique env var in the macOS part of `nightly.yml` before `4_hb_make_darktable_dmg.sh` gets called, i.e. in the previous step:

      - name: Build macOS package
        run: |
          ./src/packaging/macosx/3_make_hb_darktable_package.sh
          echo "DT_GITHUB_RUN=1" >> $GITHUB_ENV
      - name: Create DMG file
        run: |
          ./src/packaging/macosx/4_make_hb_darktable_dmg.sh

And then test for `DT_GITHUB_RUN` in `4_hb_make_darktable_dmg.sh`

Reply:
> Because default environment variables are set by GitHub and not defined in a workflow, they are not accessible through the env context.

I think you’re misinterpreting what “env context” is. Default variables should be fine to use in scripts running on runner, otherwise they would be totally useless.

Reply:
Just testet:

    steps:
      - name: Check for github runner
        run: |
          echo "${GITHUB_RUN_ID}"

result: `8002392447`

So it should be valid to use.